### PR TITLE
Anchored position: Flip to alternative alignments on overflow

### PR DESCRIPTION
**Pull request by siddharthkp**
tl;dr: If there isn't enough space  for `align: 'start'`, flip alignment to `align: 'end'` (based on side alternateOrders implementation)

| Before | After |
| ------------- |-------------|
| while in bounds, tooltip is positioned incorrectly | tooltip position is flipped |

- [See tests for more examples](https://github.com/primer/behaviors/pull/78/files#diff-966bd08d7c9fc864ddddfa7cd59d9edfa2440704b1985bf2825253e5a1b5827bR295)
- Fixes: https://github.com/primer/behaviors/issues/63
- Video explanation: https://github.rewatch.com/video/knwsmsoh21xfkv2s-2006-720p
- Tested with primer/react: https://github.com/primer/react/pull/2006

### CODE REVIEW

File: src/anchored-position.ts [hunk 1]
```diff
@@ -440,3 +476,23 @@ function shouldRecalculatePosition(
...
+  align: AnchorAlignment,
+  currentPos: Position,
+  containerDimensions: BoxPosition,
+  elementDimensions: Size
+) {
+  if (align === 'end') {
+    return currentPos.left < containerDimensions.left
+  } else if (align === 'start' || align === 'center') {
+    return currentPos.left + elementDimensions.width > containerDimensions.left + containerDimensions.width
+  }
```
[the remark this answers is not in the document]
Comment (siddharthkp) [reply src/anchored-position.ts:497]: oooh that's a great point! I understand what you mean, will play around with it a bit more

[the remark this answers is not in the document]
Comment (siddharthkp) [reply src/anchored-position.ts:497]: I tried to force the above example, does this feel like the correct behavior?

It stays `center` as long as it can, then switches to `end` if there isn't enough space.


https://user-images.githubusercontent.com/1863771/162779652-e6a8df26-7b77-4b4a-a576-03a2ebf3cb88.mov

[the remark this answers is not in the document]
Comment (siddharthkp) [reply src/anchored-position.ts:497]: > Notably, this logic for determining overflow may need to consider whether anchorSide is top or bottom, rather than left or right.

Not sure I understand when this breaks. If the anchorSide is `right` and there isn't enough space, it switches to `left` before we re-evaluate alignment. 🤔  What scenario am I missing?

**Review by rezrah — APPROVED** [review]
✨

File: src/anchored-position.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (khiga8) [reply src/anchored-position.ts:497]: I agree with you. I don't think `shouldRecalculateAlignment` would ever be called for the `anchorSide` left or right scenario 👍 If I'm understanding correctly, this method is only relevant for `anchorSide` top or bottom.

**Review by khiga8 — APPROVED** [review]
This looks great 🙌 Thank you!

File: src/anchored-position.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (siddharthkp) [reply src/anchored-position.ts:497]: Yep! It's only relevant if `anchorSide` is top or bottom 👍
